python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -644,7 +644,8 @@ def replicate(self, destSE, sourceSE=''):
         logger.info("Replicating file %s to %s" % (self.lfn, destSE))
         stdout = execute('replicateFile("%s", "%s", "%s")' % (self.lfn, destSE, sourceSE))
         if isinstance(stdout, dict) and stdout.get('OK', False) and self.lfn in stdout.get('Value', {'Successful': {}})['Successful']:
-            self.locations.append(destSE)
+            if destSE not in self.locations:
```

Comment:
That was my concern I guess. I'm happy to treat `locations` as a cache of information for the user but I worry that if it gets out of date (maybe a 'remove replica' behind our backs) then we don't want to silently do nothing. On the other hand I do agree with you about avoiding making a call to DIRAC if we can avoid it, however, this particular call (a replicate to an existing replica) it is effectively a no-op at DIRAC's end.

Really, `locations` should be a `set` in which case this wouldn't be an issue any more either way.